frontend/src/metabase-lib/lib/queries/StructuredQuery.js

```diff
@@ -351,58 +351,70 @@ export default class StructuredQuery extends AtomicQuery {
       query = query.setSourceQuery(sourceQuery.clean());
     }
 
-    query = query
+    return query
       .cleanJoins()
       .cleanExpressions()
       .cleanFilters()
       .cleanAggregations()
       .cleanBreakouts()
       .cleanSorts()
       .cleanLimit()
-      .cleanFields();
+      .cleanFields()
+      .cleanEmpty();
+  }
 
-    // remove empty queries
-    const newSourceQuery = query.sourceQuery();
-    if (newSourceQuery && (query.isEmpty() || !query.hasAnyClauses())) {
-      return newSourceQuery;
+  cleanNesting(): StructuredQuery {
```

Comment:
Pretty simple, but maybe worth a unit test.

Reply:
👍 Thanks for keeping me honest :)